packages/components/src/components/Address/Address.tsx

```diff
@@ -1,16 +1,67 @@
-import { Input } from '@chakra-ui/react';
-import React from 'react';
+import {
+  FormControl,
+  FormErrorMessage,
+  Input,
+  InputGroup,
+  InputRightElement,
+} from '@chakra-ui/react';
+import { CopyIcon, CheckIcon } from '@chakra-ui/icons';
+import React, { useState } from 'react';
 
 export interface AddressProps {
   /**
    * The address to display
    */
   value: string;
+  /**
+   * Wheter the address can be copied or not
+   */
+  copiable?: boolean;
+}
+
+interface EventTarget {
+  value: string;
+}
+
+interface SyntheticEvent {
+  currentTarget: EventTarget;
 }
 
 /**
  * A component to display an address
  */
-export const Address: React.FC<AddressProps> = ({ value }) => {
-  return <Input value={value} />;
+export const Address: React.FC<AddressProps> = ({ value, copiable = false }) => {
+  const [error, setError] = useState<null | string>(null);
+  const [copied, setCopied] = useState<boolean>(false);
+
+  const handleClick = async (event: SyntheticEvent): Promise<void> => {
+    const value = event.currentTarget.value as string;
+
+    setError(null);
```

Comment:
Why are we setting `error` to `null`?

Reply:
fixed as part of https://github.com/Developer-DAO/web3-ui/pull/105/commits/dcd5eb650f1a3e4ad0cc5c9a2479ca1894c78cef